SpeedyWeatherInternals/ext/SpeedyWeatherInternalsMetalExt.jl

```diff
@@ -0,0 +1,39 @@
+module SpeedyWeatherInternalsMetalExt
+
+    import  Metal: Metal, MtlArray, MtlDeviceArray
+    import  SpeedyWeatherInternals.Architectures: Architectures, 
+            GPU, CPU, MetalGPU, array_type, architecture, on_architecture, architecture,
+            compatible_array_types, nonparametric_type
+
+
+    # DEVICE SETUP FOR METAL
+    # extend functions from Architectures
+    Architectures.array_type(::GPU) = MtlArray
+    Architectures.array_type(::Type{<:GPU}) = MtlArray
+    Architectures.array_type(::GPU, NF::Type, N::Int) = MtlArray{NF, N, Metal.PrivateStorage}
+
+    Architectures.compatible_array_types(::GPU) = (MtlArray, MtlDeviceArray)
+    Architectures.compatible_array_types(::Type{<:GPU}) = (MtlArray, MtlDeviceArray)
+
+    Architectures.nonparametric_type(::Type{<:MtlArray}) = MtlArray
+
+    Architectures.MetalGPU() = GPU(Metal.MetalBackend())
+    Architectures.GPU() = MetalGPU() # default to Metal
+
+    Architectures.architecture(::MtlArray) = MetalGPU()
+    Architectures.architecture(::Type{<:MtlArray}) = MetalGPU()
+    Architectures.architecture(::Type{<:MtlDeviceArray}) = MetalGPU()
+
+    Architectures.on_architecture(::CPU, a::MtlArray) = Array(a)
+    Architectures.on_architecture(::GPU, a::MtlArray) = a
```

Comment:
This would need to be something like `GPU{MetalBackend}` to be on the safe side in case users load both CUDA and Metal. See also the JLArrays extension in #824.